public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx

```diff
@@ -104,50 +106,64 @@ export function FolderAndGroup({ initialFolder }: FolderAndGroupProps) {
     formState: { errors },
     watch,
     control,
-    setValue,
   } = useFormContext<RuleFormValues>();
 
   const styles = useStyles2(getStyles);
   const dispatch = useDispatch();
   const folderFilter = useRuleFolderFilter(initialFolder);
-  const [isAddingGroup, setIsAddingGroup] = useState(false);
 
   const folder = watch('folder');
   const group = watch('group');
-  const [selectedGroup, setSelectedGroup] = useState(group);
+  const [selectedGroup, setSelectedGroup] = useState<SelectableValue<string>>({ label: group, title: group });
   const initialRender = useRef(true);
 
   const { groupOptions, loading } = useGetGroupOptionsFromFolder(folder?.title ?? '');
 
-  useEffect(() => setSelectedGroup(group), [group, setSelectedGroup]);
+  useEffect(() => setSelectedGroup({ label: group, title: group }), [group, setSelectedGroup]);
 
   useEffect(() => {
     dispatch(fetchRulerRulesIfNotFetchedYet(GRAFANA_RULES_SOURCE_NAME));
   }, [dispatch]);
 
   const resetGroup = useCallback(() => {
     if (group && !initialRender.current && folder?.title) {
-      setSelectedGroup('');
+      setSelectedGroup({ label: '', title: '' });
     }
     initialRender.current = false;
   }, [group, folder?.title]);
 
-  useEffect(() => {
-    setValue('group', selectedGroup);
-  }, [selectedGroup, setValue]);
-
   const groupIsInGroupOptions = useCallback(
     (group_: string) => {
       return groupOptions.includes((groupInList: SelectableValue<string>) => groupInList.label === group_);
     },
     [groupOptions]
   );
+  const sliceResults = (list: Array<SelectableValue<string>>) => list.slice(0, SLICE_GROUP_RESULTS_TO);
+
+  const getOptions = useCallback(
+    async (query: string) => {
```

Comment:
Does the interface want this to be an async function? I'm not sure if we want to return a promise unless we have to.

Reply:
yes, we need because loadOptions requires a Promise
`loadOptions?: (inputValue: string, callback: (options: OptionsOrGroups<Option, Group>) => void) => Promise<OptionsOrGroups<Option, Group>> | void;`